spring-boot-modules/spring-boot-crud/src/test/java/com/baeldung/entity_dto_differences/controller/UserControllerIntegrationTest.java

```diff
@@ -0,0 +1,92 @@
+package com.baeldung.entity_dto_differences.controller;
+
+import com.baeldung.entity_dto_differences.dto.BookDto;
+import com.baeldung.entity_dto_differences.dto.UserCreationDto;
+import com.baeldung.entity_dto_differences.dto.UserResponseDto;
+import com.baeldung.entity_dto_differences.entity.Book;
+import com.baeldung.entity_dto_differences.entity.User;
+import com.baeldung.entity_dto_differences.repository.UserRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class UserControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    public void beforeEach() {
+
+        userRepository.deleteAll();
+    }
+
+    @Test
+    public void whenGetUsers_thenCorrect() throws Exception {
```

Comment:
Your test names do not match what is going on in the tests. If you have a given/when/then section in the test body, I would expect the test name to reflect those parts too.